cmd/sops/main.go

```diff
@@ -243,6 +252,20 @@ func decrypt(c *cli.Context, file string, fileBytes []byte, output io.Writer) er
 	return nil
 }
 
+func generateKey(tree *sops.Tree) ([]byte, error) {
```

Comment:
I think you meant tree.GenerateDataKey()? In any case, that's what I did. That function needs a tree (or Metadata) to save the generated key on all the keysources.